### PR TITLE
generic: 6.6: replace downstream patches for RealTek PHYs

### PR DESCRIPTION
**Pull request by dangowrt**
Replace downstream patches for RealTek Ethernet PHYs with the versions accepted in upstream Linux (with the exception of filling possible_interfaces which isn't supported in Linux 6.6 and a backport seems to be too involving).

Rebase all other downstream patches on top of that, except for patch "net: phy: realtek: detect early version of RTL8221B" which should no longer be needed.

### CODE REVIEW

File: target/linux/generic/backport-6.6/735-05-v6.10-net-phy-realtek-configure-SerDes-mode-for-rtl822xb-P.patch [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+[ refactored, dropped HiSGMII mode and changed commit message ]
+Signed-off-by: Marek Behún <kabel@kernel.org>
+[ changed rtl822x_update_interface() to use vendor register ]
+[ always fill in possible interfaces ]
+[ only apply to rtl8221b and rtl8226b phy's ]
+[ set phydev->rate_matching in .config_init() ]
+Signed-off-by: Eric Woudstra <ericwouds@gmail.com>
+
+Reviewed-by: Russell King (Oracle) <rmk+kernel@armlinux.org.uk>
+Reviewed-by: should come before them, without any blank lines. As the
```
Comment (Ansuel) [inline target/linux/generic/backport-6.6/735-05-v6.10-net-phy-realtek-configure-SerDes-mode-for-rtl822xb-P.patch:33]: Sorry for the noise but wth happen here AHAHAHHA

**Comment by Ansuel** [comment]
@robimarko I can remember but maybe the possible_interface thing is already backported?

**Comment by robimarko** [comment]
@Ansuel I backported it for RB5009 for 6.1 and yesterday for 6.6 as QCA808x needs it when used on DSA switch.

I can make a PR for it

File: target/linux/generic/backport-6.6/735-05-v6.10-net-phy-realtek-configure-SerDes-mode-for-rtl822xb-P.patch (hunk 1 above)
Comment (KanjiMonster) [reply target/linux/generic/backport-6.6/735-05-v6.10-net-phy-realtek-configure-SerDes-mode-for-rtl822xb-P.patch:33]: https://lore.kernel.org/netdev/417f3e08-136b-47db-880b-c80aa5698c9b@lunn.ch/

> Hi Eric
>
> The ordering of these tags should put your Signed-off-by last.
> **Reviewed-by: should come before them, without any blank lines. As the**
> patch makes it way towards mainline, each Maintainer will add a
> Signed-off: by, and it is nice to keep them together.

Patchwork saw a line starting with `Reviewed-by:` and "helpfully" merged it into the patch automatically.

**Comment by Ansuel** [comment]
> QCA808x needs it when used on DSA switch.

oh? Any device that does that?

**Comment by robimarko** [comment]
> > QCA808x needs it when used on DSA switch.
> 
> oh? Any device that does that?

RB5009 has it attached to a Marvell switch

File: target/linux/generic/backport-6.6/735-05-v6.10-net-phy-realtek-configure-SerDes-mode-for-rtl822xb-P.patch (hunk 1 above)
Comment (dangowrt) [reply target/linux/generic/backport-6.6/735-05-v6.10-net-phy-realtek-configure-SerDes-mode-for-rtl822xb-P.patch:33]: ... and it ended up in Torvald's tree like that https://github.com/torvalds/linux/commit/deb8af5243504e379878ae3f9a091b21422d65b2

Comment (Ansuel) [reply target/linux/generic/backport-6.6/735-05-v6.10-net-phy-realtek-configure-SerDes-mode-for-rtl822xb-P.patch:33]: Oh god... OK I'M DEAD AHAHAHHAHAHAHAH lets pretend nobody notice that

**Comment by Ansuel** [comment]
@robimarko sorry to ask... any news for the backport?

**Comment by robimarko** [comment]
@Ansuel https://github.com/robimarko/openwrt/commit/2b942af3fd353ee3023a391c8a16ddb562330b28

I plan to add it as part of RB5009, still need to refresh the 6.1 patches.

**Comment by Noltari** [comment]
I should have searched for this PR before I started working on the same thing xD...

Anyway, since I've already done it, maybe @dangowrt can have a look at it...
https://github.com/Noltari/openwrt/commits/realtek-phy-6.6-backports
https://github.com/Noltari/openwrt/commit/7cacfdbad585a6edfb95576aae0e2f73fd221605

BTW, there are some differences between your patches and my patches:
- MDIO_AN_10GBT_CTRL_ADV10G is still added at 727-net-phy-realtek-use-inline-functions-for-10GbE-adver.patch
- All phy_driver struct patches are done at the beginning (next to name) in order to provide better context for future upstream backports.

**Comment by namiltd** [comment]
Duplicate of https://github.com/openwrt/openwrt/commit/ca4469045f6a4845dbf8aeabd4b04b421209f66e

**Comment by Noltari** [comment]
> Duplicate of [ca44690](https://github.com/openwrt/openwrt/commit/ca4469045f6a4845dbf8aeabd4b04b421209f66e)

As @namiltd stated, my suggestion was merged by @dangowrt -> ca4469045f6a4845dbf8aeabd4b04b421209f66e

**Pull request closed without merging**